scripts/powershell/ref-updater.ps1

```diff
@@ -0,0 +1,38 @@
+# Updates the Ref in all Yml files in the Repository
+param (
+    [String]$RepoRoot,
+    [String]$Tag,
+    [String]$ToolRepo
+)
+
+Install-Module -Name powershell-yaml -RequiredVersion 0.4.2 -Force -Scope CurrentUser
+
+$ymlFiles = Get-ChildItem -Path $RepoRoot -File -Include *.yml -Recurse
+
+foreach ($file in $ymlFiles)
+{
+    $ymlContent = Get-Content $file.FullName -Raw
+    $ymlObj = ConvertFrom-Yaml $ymlContent -Ordered
+    $regex = [Regex]"ref: refs/tags/azure-sdk-tools_\d\d\d\d\d\d\d\d\.\d"
```

Comment:
Looks like you hardcoded `azure-sdk-tools` here instead of using $ToolsRepo

Reply:
If we are using regex why not just avoid the yml parser completely and just use text? I think the text you are looking for should be unique enough.

Reply:
I think I would change the regex to something like
```
$regex = [Regex]"ref: refs/tags/(?<tag>${ToolRepo}_[\d\.]+)"

$ymlContent = Get-Content $ymlContent

$updated = $false

$ymlContent = $ymlContent | Foreach-Object { 
  if ($_-match $regex) { 
   $_ -replace $matches["tag"], $Tag
   $updated = $true
  }
  else {
   $_
  }
}

if ($updated) {
  $ymlContent | Set-Content $file

```

Reply:
Looping through the object did not work to well as it created difficulties writing back the appropriate Yaml format. I instead used the regex to replace the raw string since the patter is very unique and should not exist in that form somewhere else in the yaml.